src/common/utils.tsx

```diff
@@ -161,34 +187,74 @@ export const handleValidation = (query?: string): { success: boolean; message?:
  * microToStacks
  *
  * @param {Number} amountInMicroStacks - the amount of microStacks to convert
+ * @param {Number} localString - big pretty print if true
+ */
+export const microToStacks = (
+  amountInMicroStacks: string | number,
+  localString = true
+): number | string => {
+  const value = Number(Number(amountInMicroStacks) / Math.pow(10, 6));
+  if (localString) {
+    return value.toLocaleString(undefined, { minimumFractionDigits: 2, maximumFractionDigits: 6 });
+  }
+  return value;
+};
+
+/**
+ * stacksToMicro
+ *
+ * @param {String || Number} amountInStacks - the amount of stacks to convert
  */
-export const microToStacks = (amountInMicroStacks: string | number): number =>
-  amountInMicroStacks ? Number(amountInMicroStacks) / Math.pow(10, 6) : 0;
+export const stacksToMicro = (amountInStacks: string | number) =>
+  amountInStacks ? Math.floor(Number(amountInStacks) * MICROSTACKS_IN_STACKS) : 0;
 
 export const getContractName = (fullyRealizedName: string): string =>
-  fullyRealizedName.split('.')[1];
-export const getFungibleAssetName = (assetName: string): string =>
-  getContractName(assetName).split('::')[1];
+  fullyRealizedName?.split('.')[1];
+
+export const getFunctionName = (tx: ContractCallTxs) => {
+  return tx.contract_call.function_name;
+};
+export const getFungibleAssetName = (fullyRealizedName: string): string =>
+  getContractName(fullyRealizedName)?.split('::')[1];
+
+export const getAssetNameParts = (fullyRealizedName: string) => {
+  const address = fullyRealizedName.split('.')[0];
+  const contract = getContractName(fullyRealizedName).split('::')[0];
+  const asset = getFungibleAssetName(fullyRealizedName);
+
+  return {
+    address,
+    contract,
+    asset,
+  };
+};
 
 export const getMemoString = (string: string): string | null =>
-  string ? Buffer.from(string.replace('0x', ''), 'hex').toString('utf8') : null;
+  string
+    ? Buffer.from(string.replace('0x', '').replace(/^(0{2})+|(0{2})+$/g, ''), 'hex').toString(
+        'utf8'
+      )
+    : null;
 
 export const startPad = (n: number, z = 2, s = '0'): string =>
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
   (n + '').length <= z ? ['', '-'][+(n < 0)] + (s.repeat(z) + Math.abs(n)).slice(-1 * z) : n + '';
 
 export const navgiateToRandomTx = (apiServer: string) => async () => {
   const { results } = await fetchTxList({
     apiServer: apiServer,
     types: ['smart_contract', 'contract_call', 'token_transfer'],
   })();
-  const hasNonCoinbaseTxs = results.some(tx => tx.tx_type !== 'coinbase');
+  const hasNonCoinbaseTxs = results.some((tx: any) => tx.tx_type !== 'coinbase');
 
   if (hasNonCoinbaseTxs) {
-    const nonCoinbaseResults = results.filter(tx => tx.tx_type !== 'coinbase');
+    const nonCoinbaseResults = (results as any).filter((tx: any) => tx.tx_type !== 'coinbase');
```

Comment:
If you use my other comment's fix around `fetchTxList`, then you can remove the ugly `as any`'s 😀